### PR TITLE
Add trace-cmd instrument

### PR DESCRIPTION
**Pull request by setrofim**
This pull request relies on the following devlib commits:

https://github.com/ARM-software/devlib/commit/7e80a381d8ffeb6282d476b9956d6a3d8fa63a04
https://github.com/ARM-software/devlib/commit/61b13383a4e5edbd7d369e149d3cf4d637a622c0
https://github.com/ARM-software/devlib/commit/92d9e690f03ad14279882fa8b5e7949fd36a36c9

### CODE REVIEW

File: wa/instrumentation/trace-cmd.py [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+       trace-cmd list
+
+    You may also specify ``trace_buffer_size`` setting which must be an integer
+    that will be used to set the ftrace buffer size. It will be interpreted as
+    KB::
+
+        trace_cmd_buffer_size = 8000
+
+    The maximum buffer size varies from device to device, but there is a
+    maximum and trying to set buffer size beyound that will fail. If you plan
```
Comment (marcbonnici) [inline wa/instrumentation/trace-cmd.py]: ``beyound`` is misspelt.

File: wa/instrumentation/trace-cmd.py [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+                  description="""
+                  Do not install the bundled trace-cmd  and use the one on the
+                  device instead. If there is not already a trace-cmd on the
+                  device, an error is raised.
+                  """),
+        Parameter('report_on_target', kind=bool, default=False,
+                  description="""
+                  When enabled generation of reports will be done host-side
+                  because the generated file is very large. If trace-cmd is not
+                  available on the host device this setting and be disabled and
```
Comment (marcbonnici) [inline wa/instrumentation/trace-cmd.py]: Typo: "this setting `can` be disabled. "

File: wa/instrumentation/trace-cmd.py [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+        outfile = os.path.join(context.output_directory, 'trace.dat')
+        self.collector.get_trace(outfile)
+        if self.report and not self.report_on_target:
+            textfile = os.path.join(context.output_directory, 'trace.txt')
+            self.collector.report(outfile, textfile)
+
+    def teardown(self, context):
+        self.target.remove(os.path.join(self.target.working_directory, OUTPUT_TRACE_FILE))
+        if self.report_on_target:
+            self.target.remove(os.path.join(self.target.working_directory, OUTPUT_TEXT_FILE))
```
Comment (marcbonnici) [inline wa/instrumentation/trace-cmd.py]: Should these be using `self.target.path.join`?

**Comment by setrofim** [comment]
Updated.

**Pull request merged**